mne/io/brainvision/tests/test_brainvision.py

```diff
@@ -646,7 +684,7 @@ def test_automatic_vmrk_sfreq_recovery():
 @testing.requires_testing_data
 def test_event_id_stability_when_save_and_fif_reload(tmpdir):
     """Test load events from brainvision annotations when read_raw_fif."""
-    fname = op.join(str(tmpdir), 'bv-raw.fif')
+    fname = tmpdir / 'bv-raw.fif'
```

Comment:
locally, this test failed, and I have no idea why.

Reply:
that is --> it does not fail because of my cosmetic change here, but because original_event_id and event_id are not equal. See:

```
(Pdb) event_id
{'Comma,Type/CommaValue,1': 1, 'Event/254': 2, 'New Segment/': 3, 'Optic/O  1': 4, 'Response/R255': 5, 'Stimulus/S253': 6, 'Stimulus/S255': 7, 'SyncStatus/Sync On': 8}
(Pdb) original_event_id
{'Comma,Type/CommaValue,1': 10001, 'Event/254': 254, 'New Segment/': 99999, 'Optic/O  1': 2001, 'Response/R255': 1255, 'Stimulus/S253': 253, 'Stimulus/S255': 255, 'SyncStatus/Sync On': 99998}
(Pdb) 
```

Reply:
It somehow has to do with your changes to `test.vmrk` above. Did you mean to revert that?

Reply:
No, that is a permanent change I wanted to add to the test file :thinking: but looking at the code, I also don't see why that would be a problem:

```Python
    fname = tmpdir / 'bv-raw.fif'
    raw = read_raw_brainvision(vhdr_path, eog=eog)
    original_events, original_event_id = events_from_annotations(raw)

    raw.save(fname)
    raw = read_raw_fif(fname)
    events, event_id = events_from_annotations(raw)

    assert event_id == original_event_id
    assert_array_equal(events, original_events)
```

1. we read the BV file
1. we get events from it (which works - even for the newly added event, as shown in the other test)
1. we save it to FIF
1. we read back from FIF
1. we get events from the just-read FIF annotations
1. ... they are different?

Reply:
Probably based on `event_id='auto'` default and `_check_bv_annot`, somehow your new event makes `_check_bv_annot` return False rather than True

Reply:
... to avoid needing to deal with this in this PR, I would do the same trick as above, namely don't modify the actual `test.mrk` file, but copy it line-by-line and add an event at the bottom

Reply:
ahhh got it @larsoner ! Because of these lines:

https://github.com/mne-tools/mne-python/blob/fa7064a8d4cf8857d66ca001a17fa66e7121d3bc/mne/annotations.py#L956-L963

basically, the first raw is a "brainvision raw" ... and the second raw is a "fif raw" ... and the annotation-reading with event_id="auto" depends on the kind of raw format.

So passing an event_id other than "auto" at the second events_from_annotations call should fix it.

But is that what we want? The test confuses me a bit with it's goal.

Reply:
yup, `events, event_id = events_from_annotations(raw, original_event_id)` fixes it

Reply:
oh weird, I didn't see your comments :) should have refreshed to save me some detective work

Reply:
Don't do that. The goal of the test is that "standard BV events get parsed as BV events regardless of whether it's RawBrainvision or RawFIF". Passing the event_id defeats this purpose. Hence my suggestion to use the `tmpdir` + rewrite + append approach in the one test where you actually want to check for comma support rather than modifying `test.mrk` on disk directly in this PR

Reply:
I see, and introducing non-standard events obviously breaks the thing, reasonably so. Thanks!